trestle/core/commands/replicate.py

```diff
@@ -14,11 +14,234 @@
+
+
+class TargetDefinitionCmd(Command):
+    """Replicate a target within the trestle directory structure."""
+
+    name = 'target-definition'
+
+    def _run(self, args: argparse.Namespace) -> int:
+        return ReplicateCmd.replicate_object(self.name, target.TargetDefinition, args)
+
+
+class ComponentDefinitionCmd(Command):
+    """Replicate a component definition within the trestle directory structure."""
+
+    name = 'component-definition'
+
+    def _run(self, args: argparse.Namespace) -> int:
+        return ReplicateCmd.replicate_object(self.name, component.ComponentDefinition, args)
+
+
+class SystemSecurityPlanCmd(Command):
+    """Replicate a system security plan within the trestle directory structure."""
+
+    name = 'system-security-plan'
+
+    def _run(self, args: argparse.Namespace) -> int:
+        return ReplicateCmd.replicate_object(self.name, ssp.SystemSecurityPlan, args)
+
+
+class AssessmentPlanCmd(Command):
+    """Replicate an assessment plan within the trestle directory structure."""
+
+    name = 'assessment-plan'
+
+    def _run(self, args: argparse.Namespace) -> int:
+        return ReplicateCmd.replicate_object(self.name, assessment_plan.AssessmentPlan, args)
+
+
+class AssessmentResultCmd(Command):
+    """Replicate an assessment result within the trestle directory structure."""
+
+    name = 'assessment-results'
+
+    def _run(self, args: argparse.Namespace) -> int:
+        return ReplicateCmd.replicate_object(self.name, assessment_results.AssessmentResults, args)
+
+
+class PlanOfActionAndMilestonesCmd(Command):
+    """Replicate a plan of action and milestones within the trestle directory structure."""
+
+    name = 'plan-of-action-and-milestones'
+
+    def _run(self, args: argparse.Namespace) -> int:
+        return ReplicateCmd.replicate_object(self.name, poam.PlanOfActionAndMilestones, args)
+
 
 class ReplicateCmd(Command):
-    """Replicate an existing trestle model generating new UUIDs."""
+    """Replicate an existing top level OSCAL model within the trestle project."""
 
     name = 'replicate'
+
+    subcommands = [
+        CatalogCmd,
+        ProfileCmd,
+        TargetDefinitionCmd,
+        ComponentDefinitionCmd,
+        SystemSecurityPlanCmd,
+        AssessmentPlanCmd,
+        AssessmentResultCmd,
+        PlanOfActionAndMilestonesCmd
+    ]
+
+    def _init_arguments(self) -> None:
+        logger.debug('Init arguments')
+        self.add_argument('-f', '--file', help='OSCAL model to replicate.', type=str, required=True)
+
+        self.add_argument('-o', '--output', help='Name of replicated model.', type=str, required=True)
+
+        self.add_argument(
+            '-r', '--regenerate', type=bool, default=False, help='Enable to regenerate uuids within the document'
+        )
+
+    @classmethod
+    def replicate_object(cls, model_alias: str, object_type: Type[TLO], args: argparse.Namespace) -> int:
+        """Core replicate routine invoked by subcommands."""
```

Comment:
For our docs,
I'd like to start adding google style docstrings for public function and class declarations.
In this case your comment block would look like this:
```
 """
Core replicate routine invoked by subcommands.

Args:
     model_alias: Description here. you don't need tying due to the type annotations
     object_type: description here

Returns:
      A return code that can be used as standard posix codes. 0 is success.
```
"""